### PR TITLE
Added api

### PR DESCRIPTION
**Pull request by selva-bct**
- register
 - login
 - first time password change
 - forgot password
 - change password
 - validate token

### CODE REVIEW

File: user-module/config/production.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const path = require('path')
+const fs = require('fs-extra')
+const logFilePath = path.join(__dirname, './../tmp/user-module.log')
+
+fs.ensureFileSync(logFilePath)
+
+module.exports = {
+  name: 'selva',
```
Comment (Eswar-BCT) [inline user-module/config/production.js:8]: Remove name

File: user-module/config/production.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    userPoolId: '',
+    appClientId: '',
+    poolRegion: '',
+    accessKeyId: '',
+    secretAccessKey: '',
+    apiVersion: '',
+    loggerConfig: {
+      type: 'rotating-file',
+      path: logFilePath,
+      period: '1d'
```
Comment (Eswar-BCT) [inline user-module/config/production.js:19]: make log period as one week

File: user-module/constant/constant.js [hunk 3]
```diff
@@ -33,9 +33,12 @@ export const cognitoUserAuthentication = {
...
 
-// module.exports = {
-//   defaultMessage,
-//   defaultStatusCode,
-//   cognitoUserCreation,
-//   cognitoUserAuthentication
-// }
+export const forgotPasswordRequest = {
+  SUCCESS: 'Forgot Password Request was successful. Check your email for reset link',
+  ERROR: 'Error while changing password...'
```
Comment (Eswar-BCT) [inline user-module/constant/constant.js:38]: change this error message 
`error while request for for got password`

File: user-module/controller/user-controller.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+      region: arg.poolRegion
+    })
+
+    this.cognitoClient = new AWS.CognitoIdentityServiceProvider({
+      apiVersion: arg.apiVersion,
+      region: arg.poolRegion
+    })
+  }
+
+  async authenticate (req, res) {
```
Comment (Eswar-BCT) [inline user-module/controller/user-controller.js:33]: change authenticate to login

File: user-module/package.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
     "lint": "eslint . --fix"
   },
   "engines": {
-    "node": "12.13.1"
+    "node": "12.0.0"
```
Comment (Eswar-BCT) [inline user-module/package.json:11]: change node version to 12.13.1

File: user-module/package.json [hunk 6]
```diff
@@ -27,8 +27,11 @@
     "body-parser": "1.19.0",
     "bunyan": "1.8.12",
     "config": "3.2.4",
+    "cors": "^2.8.5",
+    "es6-promisify": "^6.0.2",
```
Comment (Eswar-BCT) [inline user-module/package.json:31]: Remove this es6 line

File: user-module/config/production.js (hunk 1 above)
Comment (selva-bct) [reply user-module/config/production.js:8]: resolved

File: user-module/controller/user-controller.js (hunk 4 above)
Comment (selva-bct) [reply user-module/controller/user-controller.js:33]: Let it be as authenticate.

File: user-module/package.json (hunk 5 above)
Comment (selva-bct) [reply user-module/package.json:11]: resolved

File: user-module/package.json (hunk 6 above)
Comment (selva-bct) [reply user-module/package.json:31]: resolved

File: user-module/config/production.js (hunk 2 above)
Comment (selva-bct) [reply user-module/config/production.js:19]: resolved

File: user-module/constant/constant.js (hunk 3 above)
Comment (selva-bct) [reply user-module/constant/constant.js:38]: resolved

**Comment by selva-bct** [comment]
duplicate

**Pull request closed without merging**